### PR TITLE
Add headers for SPV_NV_shader_invocation_reorder.

### PR DESCRIPTION
**Pull request by alelenv**

### CODE REVIEW

**Comment by alelenv** [comment]
cc : @dgkoch 

@johnkslang or @dneto0 to review
Task list issue contains all related collateral

File: include/spirv/unified1/spirv.core.grammar.json [hunk 1]
```diff
@@ -4754,6 +4754,494 @@
...
+        { "kind" : "IdRef", "name" : "'SBT Record Stride'" },
+        { "kind" : "IdRef", "name" : "'Origin'" },
+        { "kind" : "IdRef", "name" : "'TMin'" },
+        { "kind" : "IdRef", "name" : "'Direction'" },
+        { "kind" : "IdRef", "name" : "'TMax'" },
+        { "kind" : "IdRef", "name" : "'Current Time'" },
+        { "kind" : "IdRef", "name" : "'HitObject Attributes'" }
+      ],
+      "capabilities" : [ "ShaderInvocationReorderNV" ],
+      "extensions" : [ "SPV_NV_shader_invocation_reorder" ],
```
Comment (dneto0) [inline include/spirv/unified1/spirv.core.grammar.json:4778]: Entries that are not capabilities should only depend on capabilities, and not extensions.
Only a capability entry should depend on an extension.

Please remove the extensions entry everywhere except for the one at line 14430.

Sorry the CI doesn't catch this.
I have an outstanding task to fix that. https://github.com/KhronosGroup/SPIRV-Headers/issues/278

**Review by dneto0 — CHANGES_REQUESTED** [review]

**Review by dgkoch — CHANGES_REQUESTED** [review]
Please also pick up the capability changes that we recently added in https://github.com/KhronosGroup/SPIRV-Registry/pull/175

(base capability depends on RayTracingKHR, and the *Motion* instructions also depend on the RayTracingMotionBlurNV capability)

File: include/spirv/unified1/spirv.core.grammar.json [hunk 2]
```diff
@@ -14427,6 +14394,7 @@
         {
           "enumerant" : "ShaderInvocationReorderNV",
           "value" : 5383,
+          "capabilties" : [ "RayTracingKHR" ],
```
Comment (dgkoch) [inline include/spirv/unified1/spirv.core.grammar.json:14397]:
```suggestion
          "capabilities" : [ "RayTracingKHR" ],
```

**Review by dgkoch — CHANGES_REQUESTED** [review]

File: include/spirv/unified1/spirv.core.grammar.json (hunk 2 above)
Comment (alelenv) [reply include/spirv/unified1/spirv.core.grammar.json]: My bad, thanks for the catch

**Review by dgkoch — COMMENTED** [review]
LGTM now.

**Review by dgkoch — COMMENTED** [review]
👍

**Review by dgkoch — APPROVED** [review]

File: include/spirv/unified1/spirv.core.grammar.json [hunk 3]
```diff
@@ -10422,6 +10877,13 @@
           "capabilities" : [ "PhysicalStorageBufferAddresses" ],
           "version" : "1.5"
         },
+        {
+          "enumerant" : "HitObjectAttributeNV",
+          "value" : 5385,
+          "extensions" : [ "SPV_NV_shader_invocation_reorder" ],
+          "capabilities" : [ "ShaderInvocationReorderNV" ],
```
Comment (alan-baker) [inline include/spirv/unified1/spirv.core.grammar.json:10884]: Think you should drop the extensions entry here too.

Comment (alelenv) [reply include/spirv/unified1/spirv.core.grammar.json:10884]: I can, but seems we have extensions present for others (like ShaderRecordBufferKHR)
Are we expected to not have any extensions?

Comment (dgkoch) [reply include/spirv/unified1/spirv.core.grammar.json:10884]: It's the same issue David pointed out earlier in https://github.com/KhronosGroup/SPIRV-Headers/pull/300#discussion_r1014039717

If there is a capability, we don't also need the extension listed.

**Review by dneto0 — APPROVED** [review]
Thanks for the updates!
LGTM.

**Pull request merged**